linux/keyman-config/km-package-install

```diff
@@ -137,17 +140,18 @@ def main():
         if not kbdata:
             logging.error("km-package-install: error: Could not download keyboard data for %s", args.package)
             sys.exit(3)
-        if installed_kmp_ver:
-            if kbdata['version'] == installed_kmp_ver:
+        kbdata_version = secure_lookup(kbdata, 'version')
+        if installed_kmp_ver and kbdata_version:
+            if kbdata_version == installed_kmp_ver:
                 logging.error(
                     "km-package-install: The %s version of the %s keyboard package is already installed.",
                     installed_kmp_ver, args.package)
                 sys.exit(1)
-            elif float(kbdata['version']) > float(installed_kmp_ver):
+            elif float(kbdata_version) > float(installed_kmp_ver):
```

Comment:
I was not aware that version numbers were being cast as floats -- that's not safe as version number strings can contain multiple sections, e.g. `1.0.1`.